tests/run/pure_fused.pxd

```diff
@@ -0,0 +1,6 @@
+ctypedef fused NotInPy:
```

Comment:
There is also a second case that might be worth testing here: a fused type that is declared directly in the `.py` file with `cython.fused_type()`. IIRC, that should use different code paths for applying the types. (If that fails, though, it's ok to fix it in a separate PR.)